network/containerizer/bridgepolicy.go

```diff
@@ -188,6 +189,25 @@ var skippedDeviceNames = set.NewStrings(
 	network.DefaultKVMBridge,
 )
 
+// The general policy is to:
+// 1. Add br- to device name (to keep current behaviour), if it doesn fit in 15 characters then:
+// 2. Add b- to device name, if it doesn't fit in 15 characters then:
+// 3a. For devices starting in 'en' remove 'en' and add 'b-'
+// 3b. For all other devices 'b-' + 6-char hash of name + '-' + last 6 chars of name
+func BridgeNameForDevice(device string) string {
+	switch {
+	case len(device) < 13:
+		return fmt.Sprintf("br-%s", device)
+	case len(device) == 13:
+		return fmt.Sprintf("b-%s", device)
+	case device[:2] == "en":
+		return fmt.Sprintf("b-%s", device[2:])
+	default:
+		hash := crc32.Checksum([]byte(device), crc32.IEEETable) & 0xffffff
```

Comment:
I don't think we'll find someone evil enough to create collisions in network interface names :) If that'll be a problem we can always switch, but for now let's not include additional dependency.